### PR TITLE
[RFC] tests: Migrate legacy test 69.

### PR DESCRIPTION
**Pull request by lucc**

### CODE REVIEW

**Comment by Pyrohh** [comment]
@fwalch  @tarruda the build failure is interesting:
```
0.00s$ .ci/script.sh

.ci/script.sh: eval: line 17: syntax error near unexpected token `newline'
```

I'm a little confused, as the line in question is an empty line:
```
16  eval "$(curl -Ss https://raw.githubusercontent.com/neovim/bot-ci/master/scripts/travis-setup.sh) deps-x64"
17
18  sudo -E su "${USER}" -c ".ci/run_tests.sh"
```

Perhaps it's just curl(1) failing to download?

**Comment by lucc** [comment]
The test failures are [here](https://travis-ci.org/neovim/neovim/builds/70386246). I will no rebase to resolve a merge conflict so the test will be restarted.

**Comment by lucc** [comment]
ref: part of #1286

**Review by thebrianbug — APPROVED** [review]
Disclaimer: my fist time doing this EVER
TL;DR
See Works, Invalid params, Arithmetic overflow, and Edge cases entries.
- Travis: good
- Style: introduces no new errors to lint. Unfortunately the same can't be said for the rest of the code base.
- Works: passes the unit test :) appears to be equivalent.
- Understandable: 150% improvement
- Duplication: nope
- Unused vars: nope
- Modular: it's a unit test. It's also much more modular than the original
- Global/static vars: nope
- Function attribute: New to me; I don't think these apply here though? C/C++ mainly??
- Naming: Great!
- Lib functs: nope
- Debug/Logs: nope
- Loops: nope
- Return value checked: yea unit tests! Definitely checking.
- Invalid params: Have we tested this enough?? We've translated the existing test, but is there anything we want to add here??
- Use after free's: n/a
- Leaks: better not be. Not by the test itself
- &NULL: nope
- Uninitialized vars: nope
- Arithmetic overflow: Not unless byteidx and byteidxcomp can overflow
- Unneeded assert: no
- Memory usage: n/a
- Optimizations: no, this is a direct translation

## \- Unit tests: Definitely!
- Superfluous comments: Good
- Needed comments: I think we're good here. Rely on the name of the unit test for the commentary.
- Outdated comments: no
- Edge cases: this is a direct translation of what was already there. Hopefully, it's sufficient.
- Complex algorithms explained: n/a
- Wierd external lib functionality documentation: n/a
- Data structs/typedefs explained: n/a
- Incomplete code: no

**Comment by lucc** [comment]
@bmcilw1 where did you get the list from?

**Comment by bmcilw1** [comment]
Here: https://github.com/neovim/neovim/wiki/Code-review-checklist

I noticed after I did it that no one else uses it exhaustively, if it's wanted I can use markdown checklist format if not then I can omit the full list in future.

**Comment by bmcilw1** [comment]
@lucc if it's wanted it's possible that the list can be imported automatically when opening a new list review.

**Comment by justinmk** [comment]
Merged. Thank you @lucc
